fixed status not showing upon bot startup

EndlessE opened the pull request:
Made sure UpdateStatus function is called once after bot is started up using global boolean variable.

EndlessE committed:
fixed status not showing upon bot startup

Made sure UpdateStatus function is called once after bot is started up using global boolean variable.

zeroZshadow commented on main.go:
```diff
@@ -83,10 +86,20 @@ func connectDiscordBot() (*discordgo.Session, error) {
+	
+	if first == true {
+		first = false
+		LastInfo = *info
```
This does some heavy code duplication.

Why not instead alter `if LastInfo.Players == info.Players {` to be `if LastInfo.Players == info.Players || first {` to avoid the duplication? Just always set `first` to false at the end of the function.

EndlessE replied:
Because the suggested statement will evaluate as true as soon as ```LastInfo.Players == info.Players``` evaluates as true, which is exactly what happens at startup anyway (if it didn't, there would be no need for the ```initializedPresence``` variable. Proper solution should be ```if LastInfo.Players == info.Players && initializedPresence``` without ever resetting the variable to false.

zeroZshadow commented on main.go:
```diff
@@ -20,6 +20,8 @@ var (
 	LastInfo steam.InfoResponse
 )
 
+var first bool
```
This does not describe what its for. Renaming to `initializedPresence` would suit better.
Also, since the default value of a bool is false, and we invert the meaning with this name, the initialization to true in `main` would no longer be required.

zeroZshadow commented on main.go:
```diff
@@ -33,6 +35,7 @@ func init() {
 
 func main() {
 	var err error
+	first = true
```
This can be removed. See comment on declaration

zeroZshadow commented on main.go:
```diff
@@ -83,10 +86,20 @@ func connectDiscordBot() (*discordgo.Session, error) {
 
 func presenceUpdate(s *discordgo.Session, m *discordgo.PresenceUpdate) {
 	info, err := Server.Info()
+	
```
Why was a newline added here?

EndlessE replied:
Thought the code looked more readable that way, doesn't really matter.

EndlessE committed:
Removed duplicate code without impacting functionality

Changed variable name from "first" to "initializedPresence", made existing code call UpdateStatus at startup using said variable
